src/StoreApi/Routes/AbstractRoute.php

```diff
@@ -70,17 +70,69 @@ public function get_response( \WP_REST_Request $request ) {
 					$response = $this->get_route_response( $request );
 					break;
 			}
-			if ( 'GET' !== $request->get_method() && ! is_wp_error( $response ) ) {
-				$response->header( 'X-WC-Store-API-Nonce', wp_create_nonce( 'wc_store_api' ) );
-			}
 		} catch ( RouteException $error ) {
 			$response = $this->get_route_error_response( $error->getErrorCode(), $error->getMessage(), $error->getCode(), $error->getAdditionalData() );
 		} catch ( \Exception $error ) {
 			$response = $this->get_route_error_response( 'unknown_server_error', $error->getMessage(), 500 );
 		}
+
+		if ( is_wp_error( $response ) ) {
+			$response = $this->error_to_response( $response );
+		}
+
+		return $this->add_response_headers( $response );
```

Comment:
Yes; we should include the response in all Store API responses because:

a) It gives the client consistent headers to capture through a middleware
b) A login can occur outside of the Store API flow, so theoretically the nonce could refresh at any time (we may need a retry mechanism at some point)

Reply:
Explanation makes sense. I just want to make sure we're not exposing an easy way for bad actors to grab nonces for malicious use  - I don't think they can because the request is still tied to either a cart session or an authenticated request.